kork/kork-artifacts/src/main/java/com/netflix/spinnaker/kork/artifacts/artifactstore/s3/S3ArtifactStoreStorer.java

```diff
@@ -144,33 +147,11 @@ private byte[] getReferenceAsBytes(Artifact artifact) {
    * dynamodb.
    */
   private boolean objectExists(ArtifactReferenceURI uri) {
-    HeadObjectRequest request = HeadObjectRequest.builder().bucket(bucket).key(uri.paths()).build();
-    try {
-      s3Client.headObject(request);
-      log.debug("Artifact exists. No need to store. reference={}", uri.uri());
-      return true;
-    } catch (NoSuchKeyException e) {
-      // pretty gross that we need to use exceptions as control flow, but the
-      // java SDK doesn't have any other way of check if an object exists in s3
-      log.info("Artifact does not exist reference={}", uri.uri());
-      return false;
-    } catch (S3Exception e) {
-      int statusCode = e.statusCode();
-      log.error(
-          "Artifact store failed head object request statusCode={} reference={}",
-          statusCode,
-          uri.uri());
-
-      if (statusCode != 0) {
-        // due to this being a HEAD request, there is no message giving a clear
-        // indication of what failed. Rather than seeing a useful message back
-        // to gate, we instead see just null. To alleviate this, we wrap the
-        // exception with a more meaningful message
-        throw new SpinnakerException(buildHeadObjectExceptionMessage(e), e);
-      }
-
-      throw new SpinnakerException("S3 head object failed", e);
-    }
+    ListObjectsV2Request request =
+        ListObjectsV2Request.builder().bucket(bucket).prefix(uri.paths()).maxKeys(1).build();
+
+    ListObjectsV2Response resp = s3Client.listObjectsV2(request);
```

Comment:
NIT:  Don't need the object declaration here.  Not a blocker, just... a note.  ALSO hasContents doesn't mean the object ACTUALLY exists.  https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/s3/model/ListObjectsV2Response.html#hasContents()  probably should do a resp.getContents().isEmpty() check as well.